week6/nangniya/nangniya/src/FilterableProductTable.js

```diff
@@ -0,0 +1,54 @@
+import React, { useState } from "react";
+import SearchBar from "./SearchBar";
+import ProductTable from "./ProductTable";
+import InputBar from "./InputBar";
+
+const FilterableProductTable = ({ products, setProducts }) => {
+  const [filterText, setFilterText] = useState("");
+  const [inStockOnly, setInStockOnly] = useState(false);
+
+  const addProduct = (newProduct) => {
+    setProducts((previousData) => [...previousData, newProduct]);
+    // products = [{}, {}, {}, {}, ..., {}]
+    // newProducts = [...products, { newProduct }]
+  };
+
+  const deleteProduct = (targetProduct) => {
+    setProducts((prev) =>
+      prev.filter((product) => product.name !== targetProduct)
+    );
+  };
```

Comment:
```suggestion
  const deleteProduct = (targetProductName) => {
    setProducts((prev) =>
      prev.filter((product) => product.name !== targetProductName)
    );
  };
```

이름을 비교하는 것이기 때문에 `targetProductName`이라고 정확하게 명시하는게 좋아 보입니다!

변수 또는 함수의 이름이 길어지는 것을 걱정하실 필요는 없습니다.

길더라도 그 목적이 잘 나타나는 변수명이 훨씬 더 좋야요!